src/mixed-line-bar-chart/hooks/use-navigation.ts

```diff
@@ -80,10 +81,21 @@ export function useNavigation<T extends ChartDataTypes>({
     if (isGroupNavigation) {
       onBarGroupFocus();
     } else {
-      onLineGroupFocus();
+      onLineFocus();
     }
   };
 
+  const allUniqueX = useMemo(() => {
+    const result = [];
+    for (let i = 0; i < scaledSeries.length; i += 1) {
+      const point = scaledSeries[i];
+      if (point !== undefined && (!result.length || result[result.length - 1].scaledX !== point.x)) {
```

Comment:
Yes, we are assuming that points with the same `x` will be consecutive. I think we can assume that since `scaledSeries` is sorted by `x` as it is generated by the parent component `ChartContainer` by calling `makeScaledSeries`, which performs the sorting here:  https://github.com/cloudscape-design/components/blob/main/src/mixed-line-bar-chart/make-scaled-series.ts#L96

Reply:
Added a comment in a new commit